### ISSUE TITLE
WarmUpLR may fail to reach the configured `base_lr` when `warmup > 1`

### ISSUE THREAD

**Issue by xryami** [lingxitong/MIL_BASELINE#47]
Hello,

While training with `MIL_BASELINE`, I noticed that when `Model.scheduler.warmup > 1`, the actual learning rate does not align with the YAML configuration. I would like to ask if this is the expected behavior, a known issue, or if it has already been fixed in another branch.

### Configuration

* `lr: 0.0002`
* `warmup: 2`
* `StepLR(step_size=3, gamma=0.9)`

### Actual Observations

* **Epochs 1–4:** The training learning rate remains at `0.000100` and never rises to the configured `0.000200`.
* **Epoch 5 onwards:** `StepLR` begins its decay based on the incorrect value: `0.000090 = 0.0001 × 0.9` (instead of the expected `0.00018 = 0.0002 × 0.9`).

### Relevant Code

In `utils/model_utils.py`, inside `WarmUpLR.get_lr()`:

```python
return [self.init_lr * epoch for epoch in range(1, self.warmup_epochs + 1)]

```
